File: Model.Tests/MatchGenerators/GeneticMatchGeneratorTests.vb
<TestClass>
Public Class GeneticMatchGeneratorTests

    <TestMethod>
    Public Sub Generate2Test()
        Dim sut = New GeneticMatchGenerator(2)
        Dim result = sut.Generate()
        Assert.AreEqual(1, result.Count)
    End Sub

    <TestMethod>
    Public Sub Generate3Test()
        Dim sut = New GeneticMatchGenerator(3)
        Dim result = sut.Generate()
        Assert.AreEqual(3, result.Count)
    End Sub

    <TestMethod>
    Public Sub Generate5Test()
        Dim sut = New GeneticMatchGenerator(5)
        Dim result = sut.Generate()
        Assert.AreEqual(10, result.Count)
    End Sub

End Class

File: POFF.Kicker.Tests/MatchGenerators/GeneticMatchGeneratorTests.cs
using System.Linq;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using POFF.Kicker.MatchGenerators;

namespace POFF.Kicker.Tests.MatchGenerators;

[TestClass]
public class GeneticMatchGeneratorTests
{

    [TestMethod]
    public void Generate2Test()
    {
        var sut = new GeneticMatchGenerator(2);
        var result = sut.Generate();
        Assert.AreEqual(1, result.Count());
    }

    [TestMethod]
    public void Generate3Test()
    {
        var sut = new GeneticMatchGenerator(3);
        var result = sut.Generate();
        Assert.AreEqual(3, result.Count());
    }

    [TestMethod]
    public void Generate5Test()
    {
        var sut = new GeneticMatchGenerator(5);
        var result = sut.Generate();
        Assert.AreEqual(10, result.Count());
    }

}